kurbo/src/bezpath.rs

```diff
@@ -393,6 +393,17 @@ impl BezPath {
         cbox.unwrap_or_default()
     }
 
+    /// Get the last point of the path, if path is not empty.
+    pub fn last_point(&self) -> Option<Point> {
+        match self.0.last()? {
+            PathEl::MoveTo(p) => Some(*p),
+            PathEl::LineTo(p1) => Some(*p1),
+            PathEl::QuadTo(_, p2) => Some(*p2),
+            PathEl::CurveTo(_, _, p3) => Some(*p3),
+            PathEl::ClosePath => segments(self).last_point(),
```

Comment:
```suggestion
            PathEl::ClosePath => self.0.iter().rev().skip(1).take_while(|el| !matches(|el|, PathEl::ClosePath).last().expect("can't start a segment on a ClosePath").end_point().unwrap()
```

Ok, I'm sure this can be clearer. But the point is that if we have a long `BezPath` containing multiple `ClosePath`s then `segments` will need to go through all of them but in fact we only need to look at the last closed path.

Reply:
I think the Segments iterator will panic in that case anyway.